src/tests/Test1SortingOntheInventoryPage.spec.js

```diff
@@ -0,0 +1,53 @@
+import { expect } from '@playwright/test';
+import { test } from '../fixtures/base';
+import credentials from '../tests/credentials.json';
+
+test.describe('Saucedemo app basic tests', () => {
+    test('should sort items', async (
+        /** @type {{ app: import('../pages/Application').Application }} */{ app },
+    ) => {
+        await app.login.navigate();
+        await app.login.performLogin(credentials.standard_user.username, credentials.standard_user.password);
+
+        // verifying sorting from high to low
+        await app.inventory.sortingItems('hilo');
+        const prices = await app.inventory.getAllPrices();
+        const priceWithoutDollar = prices.map(price => parseFloat(price.replace('$', '').trim()));
+        console.log(priceWithoutDollar);
+
+        const sortedPrices = [...priceWithoutDollar].sort((a, b) => b - a);
+        console.log(sortedPrices);
+        expect(priceWithoutDollar).toEqual(sortedPrices);
+
+        // verifying sorting from low to hight
+        await app.inventory.sortingItems('lohi');
+
+        const prices1 = await app.inventory.getAllPrices();
+        const priceWithoutDollar1 = prices1.map(price => parseFloat(price.replace('$', '').trim()));
+        console.log(priceWithoutDollar1);
+
+        const sortedPrices1 = [...priceWithoutDollar1].sort((a, b) => a - b);
+        console.log(sortedPrices1);
+        expect(priceWithoutDollar1).toEqual(sortedPrices1);
+
+        // verifying sorting from za
+        await app.inventory.sortingItems('za');
+        const names = await app.inventory.getAllNames();
+        const sortedNames = names.map(name => name.trim());
+        console.log(sortedNames);
+
+        const expectedNames = [...sortedNames].sort((a, b) => b.localeCompare(a));
+        console.log(sortedNames);
+        expect(sortedNames).toEqual(expectedNames);
+
+        // verifying sorting from az
+        await app.inventory.sortingItems('az');
+        const names1 = await app.inventory.getAllNames();
+        const sortedNames1 = names1.map(name => name.trim());
+        console.log(sortedNames1);
+
+        const expectedNames1 = [...sortedNames1].sort((a, b) => a.localeCompare(b));
+        console.log(sortedNames1);
+        expect(sortedNames1).toEqual(expectedNames1);
```

Comment:
Cool! 
Now you have a test that verifies all sorting options.
The problem with such tests is if something goes wrong you don't know right from it's name what exact sorting is broken.

Let's split this test into 4 separate tests.
Once it's done, let's group it into 2 groups - sorting by name and by price - and use https://playwright.dev/docs/test-parameterize to have code for 2 cases but it will generate 4 tests in the report for each sorting option.
Feel free to tag me in chat or DM if you have more questions on it, but please prior to it - try to implement by yourself :) Deal?

Reply:
@JuliaGitt give you a hint

The Inventory.page.js file:

```
// @ts-check
import { test, expect } from '@playwright/test';
import { getRandomLocators, randomIntFromInterval } from '../utils/random_element';
import { BasePage } from './Base.page';

export class InventoryPage extends BasePage {
  url = '/inventory.html';

  sortSelect = this.page.getByTestId('product-sort-container');

  root = this.page.getByTestId('inventory-list');

  inventoryItems = this.root.getByTestId('inventory-item');

  inventoryItemTitle = this.root.getByTestId('inventory-item-name');

  inventoryItemPrice = this.root.getByTestId('inventory-item-price');

  addItemToCartButton = this.root.locator('[id^="add-to-cart"]');

  /**
   * @param {number} id
   */
  async addItemToCartById(id) {
    await this.addItemToCartButton.nth(id).click();
  }

  /**
   * @param {string} type
   */
  async selectSortOption(type) {
    await this.sortSelect.selectOption({ label: `${type}` });
  }

  async getProductsNames() {
    const productNames = await this.inventoryItemTitle.allInnerTexts();
    return productNames.map(title => title.trim());
  }

  async getProductPrices() {
    const productPrices = await this.inventoryItemPrice.allInnerTexts();
    return productPrices.map(price => parseFloat(price.replace('$', '').trim()));
  }

  /**
   * @param {string} sortBy
   * @param {string[]} productNames
   * @param {number[]} productPrices
   */
  async expectSortedProducts(sortBy, productNames, productPrices) {
    switch (sortBy) {
      case 'Name (A to Z)':
        const sortedNamesAZ = await this.getProductsNames();
        const expectedNamesAZ = productNames.sort();
        expect(sortedNamesAZ).toEqual(expectedNamesAZ);
        break;
      case 'Name (Z to A)':
        const sortedNamesZA = await this.getProductsNames();
        const expectedNamesZA = productNames.sort().reverse();
        expect(sortedNamesZA).toEqual(expectedNamesZA);
        break;
      case 'Price (low to high)':
        const sortedPricesLowHigh = await this.getProductPrices();
        const expectedPricesLowHigh = productPrices.sort((a, b) => a - b);
        expect(sortedPricesLowHigh).toEqual(expectedPricesLowHigh);
        break;
      case 'Price (high to low)':
        const sortedPricesHighLow = await this.getProductPrices();
        const expectedPricesHighLow = productPrices.sort((a, b) => b - a);
        expect(sortedPricesHighLow).toEqual(expectedPricesHighLow);
        break;
      default:
        throw new Error('Sort option is not correct');
    }
  }
}
```

Reply:
The sort.inventories.spec.js file:

```
test.describe('Sort Inventories', () => {
  const dataSortType = [
    {
      sortBy: 'Name (A to Z)',
    },
    {
      sortBy: 'Name (Z to A)',
    },
    {
      sortBy: 'Price (low to high)',
    },
    {
      sortBy: 'Price (high to low)',
    },
  ];

  dataSortType.forEach(({ sortBy }) => {
    test(`Sort products by ${sortBy}`, async (/** @type {{ app: import('../pages/Application').Application }} */ {
      app,
    }) => {
      await app.login.navigate();
      await app.login.performLogin(credentials.standard_user.username, credentials.standard_user.password);

      const productNames = await app.inventory.getProductsNames();
      const productPrices = await app.inventory.getProductPrices();

      await app.inventory.selectSortOption(sortBy);

      await app.inventory.expectSortedProducts(sortBy, productNames, productPrices);
    });
  });
});
```